src/waveform/renderers/waveformmark.h

```diff
@@ -94,23 +96,31 @@ class WaveformMark {
     // Check if a point (in image coordinates) lies on drawn image.
     bool contains(QPoint point, Qt::Orientation orientation) const;
 
+    QImage generateImage(float breath, float devicePixelRatio);
+
     QColor m_textColor;
     QString m_text;
     Qt::Alignment m_align;
     QString m_pixmapPath;
 
-    float m_linePosition;
+    float m_linePosition{};
+    int m_breadth{};
 
     WaveformMarkLabel m_label;
 
   private:
     std::unique_ptr<ControlProxy> m_pPositionCO;
     std::unique_ptr<ControlProxy> m_pEndPositionCO;
     std::unique_ptr<ControlProxy> m_pVisibleCO;
-    std::unique_ptr<QOpenGLTexture> m_pTexture; // used by allshader::WaveformRenderMark
+
     friend class allshader::WaveformRenderMark;
+
+    std::unique_ptr<QOpenGLTexture> m_pTexture; // Used by allshader::WaveformRenderMark
+    QImage m_image;                             // Used by WaveformRenderMark
+    bool m_needsUpdate{true};                   // To indicate either the texture or the image has
```

Comment:
while I don't have any objection to this personally, the pre-established style was to always initialize all members only in the member initializer list of the constructor(s).

Reply:
Ok, I wasn't aware of that. I was under the impression that the general push was to go for modern c++. Do you think this is open for discussion? Otherwise I will change it.

Reply:
I'm in favor of it, but you'll have to discuss with "the keeper of the styleguide" @daschuer. Do you have any references that this style is preferred over ours in modern c++?

Reply:
not relevant anymore